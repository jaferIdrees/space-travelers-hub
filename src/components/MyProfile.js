import React from 'react';
import { useSelector } from 'react-redux';
import Card from 'react-bootstrap/Card';
import ListGroup from 'react-bootstrap/ListGroup';
import Row from 'react-bootstrap/Row';

const MyProfile = () => {
  const rockets = useSelector((state) => state.rockets);
  const myRockets = rockets.filter((rocket) => rocket.reserved);
  const missions = useSelector((state) => state.missions);
  const myMissions = missions.filter((item) => item.joined === true);
  return (
    <>
      <link
        rel="stylesheet"
        href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.0-beta1/dist/css/bootstrap.min.css"
        integrity="sha384-0evHe/X+R7YkIZDRvuzKMRqM+OrBnVFBL6DOitfPri4tjfHxaWutUpFmBp4vmVor"
        crossOrigin="anonymous"
      />
      <Row>
        <Card style={{ width: '18rem' }} className="ms-5">
          <Card.Title>My Missions</Card.Title>
          <ListGroup variant="flush">
            {myMissions.map((mission) => (
              <ListGroup.Item key={mission.mission_id}>
                {mission.mission_name}
              </ListGroup.Item>
            ))}
          </ListGroup>
        </Card>
        <Card style={{ width: '18rem' }} className="ms-auto me-5">
          <Card.Title>My Rockets</Card.Title>
          <ListGroup variant="flush">
            {myRockets.map((rocket) => (
              <ListGroup.Item key={rocket.id}>
                {rocket.rocket_name}
              </ListGroup.Item>
            ))}
          </ListGroup>
        </Card>
      </Row>
    </>
  );
};

export default MyProfile;
