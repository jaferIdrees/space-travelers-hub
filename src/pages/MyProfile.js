import React from 'react';
import Myprofile from '../components/MyProfile';

const MyProfile = () => (
  <Myprofile />
);

export default MyProfile;
