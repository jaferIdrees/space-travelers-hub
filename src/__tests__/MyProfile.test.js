import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import MyProfile from '../components/MyProfile';

describe('MyProfile component testing using snapshots', () => {
  it('Test MyProfile component renders', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MyProfile />
        </Provider>,
      );
    expect(tree).toMatchSnapshot();
  });
});
