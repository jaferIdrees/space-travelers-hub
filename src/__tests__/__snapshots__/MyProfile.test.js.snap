// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My component testing using snapshots Test Rockets component renders 1`] = `
Array [
  <link
    crossOrigin="anonymous"
    href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.0-beta1/dist/css/bootstrap.min.css"
    integrity="sha384-0evHe/X+R7YkIZDRvuzKMRqM+OrBnVFBL6DOitfPri4tjfHxaWutUpFmBp4vmVor"
    rel="stylesheet"
  />,
  <div
    className="row"
  >
    <div
      className="ms-5 card"
      style={
        Object {
          "width": "18rem",
        }
      }
    >
      <div
        className="card-title h5"
      >
        My Missions
      </div>
      <div
        className="list-group list-group-flush"
        onKeyDown={[Function]}
      />
    </div>
    <div
      className="ms-auto me-5 card"
      style={
        Object {
          "width": "18rem",
        }
      }
    >
      <div
        className="card-title h5"
      >
        My Rockets
      </div>
      <div
        className="list-group list-group-flush"
        onKeyDown={[Function]}
      />
    </div>
  </div>,
]
`;

exports[`MyProfile component testing using snapshots Test MyProfile component renders 1`] = `
Array [
  <link
    crossOrigin="anonymous"
    href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.0-beta1/dist/css/bootstrap.min.css"
    integrity="sha384-0evHe/X+R7YkIZDRvuzKMRqM+OrBnVFBL6DOitfPri4tjfHxaWutUpFmBp4vmVor"
    rel="stylesheet"
  />,
  <div
    className="row"
  >
    <div
      className="ms-5 card"
      style={
        Object {
          "width": "18rem",
        }
      }
    >
      <div
        className="card-title h5"
      >
        My Missions
      </div>
      <div
        className="list-group list-group-flush"
        onKeyDown={[Function]}
      />
    </div>
    <div
      className="ms-auto me-5 card"
      style={
        Object {
          "width": "18rem",
        }
      }
    >
      <div
        className="card-title h5"
      >
        My Rockets
      </div>
      <div
        className="list-group list-group-flush"
        onKeyDown={[Function]}
      />
    </div>
  </div>,
]
`;

exports[`MyProfile component testing using snapshots Test Rockets component renders 1`] = `
Array [
  <link
    crossOrigin="anonymous"
    href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.0-beta1/dist/css/bootstrap.min.css"
    integrity="sha384-0evHe/X+R7YkIZDRvuzKMRqM+OrBnVFBL6DOitfPri4tjfHxaWutUpFmBp4vmVor"
    rel="stylesheet"
  />,
  <div
    className="row"
  >
    <div
      className="ms-5 card"
      style={
        Object {
          "width": "18rem",
        }
      }
    >
      <div
        className="card-title h5"
      >
        My Missions
      </div>
      <div
        className="list-group list-group-flush"
        onKeyDown={[Function]}
      />
    </div>
    <div
      className="ms-auto me-5 card"
      style={
        Object {
          "width": "18rem",
        }
      }
    >
      <div
        className="card-title h5"
      >
        My Rockets
      </div>
      <div
        className="list-group list-group-flush"
        onKeyDown={[Function]}
      />
    </div>
  </div>,
]
`;

exports[`Rockets component testing using snapshots Test Rockets component renders 1`] = `
Array [
  <link
    crossOrigin="anonymous"
    href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.0-beta1/dist/css/bootstrap.min.css"
    integrity="sha384-0evHe/X+R7YkIZDRvuzKMRqM+OrBnVFBL6DOitfPri4tjfHxaWutUpFmBp4vmVor"
    rel="stylesheet"
  />,
  <div
    className="row"
  >
    <div
      className="ms-5 card"
      style={
        Object {
          "width": "18rem",
        }
      }
    >
      <div
        className="card-title h5"
      >
        My Missions
      </div>
      <div
        className="list-group list-group-flush"
        onKeyDown={[Function]}
      />
    </div>
    <div
      className="ms-auto me-5 card"
      style={
        Object {
          "width": "18rem",
        }
      }
    >
      <div
        className="card-title h5"
      >
        My Rockets
      </div>
      <div
        className="list-group list-group-flush"
        onKeyDown={[Function]}
      />
    </div>
  </div>,
]
`;
